### PR TITLE
Fix ascii fast path optimization for functions with fixed width primitive output type

### PR DESCRIPTION
**Pull request by PHILO-HE**
Fix https://github.com/facebookincubator/velox/issues/5792.

### CODE REVIEW

**Review by laithsakka — APPROVED** [review]

**Comment by PHILO-HE** [comment]
@laithsakka, there is a check failure. Could you help check whether it is related to this PR? Thanks!

File: velox/expression/SimpleFunctionAdapter.h [hunk 1]
```diff
@@ -600,15 +600,23 @@ class SimpleFunctionAdapter : public VectorFunction {
...
-        applyContext.applyToSelectedNoThrow([&](auto row) INLINE_LAMBDA {
-          // Passing a stack variable have shown to be boost the performance
-          // of functions that repeatedly update the output. The opposite
-          // optimization (eliminating the temp) is easier to do by the
-          // compiler (assuming the function call is inlined).
-          typename return_type_traits::NativeType out{};
-          bool notNull = doApplyNotNull<0>(row, out, readers...);
-          writeResult(row, notNull, out);
-        });
+        if (applyContext.allAscii) {
```
Comment (laithsakka) [inline velox/expression/SimpleFunctionAdapter.h:603]: @PHILO-HE  internal meta tools reported build time and size regressions can you wrap this part in a constexpr
like this:
this way the compilation of the added loop will be triggrered only when callAscii is defined, would be nice to do the same for the other usage. t
```
 else if (allNotNull) {
        if constexpr (FUNC::has_ascii) {
          if (applyContext.allAscii) {
            applyContext.applyToSelectedNoThrow([&](auto row) INLINE_LAMBDA {
              typename return_type_traits::NativeType out{};
              bool notNull = doApplyAsciiNotNull<0>(row, out, readers...);
              writeResult(row, notNull, out);
            });
            return;
          }
        }

        applyContext.applyToSelectedNoThrow([&](auto row) INLINE_LAMBDA {
          // Passing a stack variable have shown to be boost the performance
          // of functions that repeatedly update the output. The opposite
          // optimization (eliminating the temp) is easier to do by the
          // compiler (assuming the function call is inlined).
          typename return_type_traits::NativeType out{};
          bool notNull = doApplyNotNull<0>(row, out, readers...);
          writeResult(row, notNull, out);
        });
      } 
      ```

Comment (PHILO-HE) [reply velox/expression/SimpleFunctionAdapter.h]: Just added the check for `has_ascii` for the two usages. Thanks!

**Comment by PHILO-HE** [comment]
Hi @laithsakka, all checks have passed. If you have any other comment, please kindly let me know. Thanks!

**Comment by laithsakka** [comment]
thanks!

**Comment by PHILO-HE** [comment]
Hi @laithsakka, there is still an internal check warning newly reported. Could you please check again?

**Pull request closed without merging**